### ISSUE TITLE
Setting LuaTable metatable __index to point to C# class instance 

### ISSUE THREAD

**Issue by Halsys** [NLua/NLua#188]
I'm having a hell of a hard time trying to wrap a table to a C# class. The ideal thing I came up with is this.

``` C#
var Code = " return function(intstance) " +                 // Create the function that will do our work
    "local Meta = { __index = instance } " +                // Create metatable 
    ScriptFile.FileName + " = setmetatable({}, Meta ) " +   // Force inheritance on to table from intstance
    ScriptFile.String() +                                   // User Code
    " return " + ScriptFile.FileName + " end ";             // Return the table and end the function

LuaFunction CodeCreation = MainState.DoString(Code, ScriptFile.FileName)[0] as LuaFunction;
Reference = CodeCreation.Call(this)[0] as LuaTable;
```

Except the metatable index just does not work... Like at all. Maybe this is something I just can't do.
When I try to access the table to retrieve something that I know exists on instance, I get an error.
Any tips would be welcome! Thanks!

**Comment by viniciusjarina** [comment]
Hi @Halsys what exactly are you trying to to :smile:  ?

**Comment by Halsys** [comment]
I'm making a C# game engine using NLua for scripting. I have a C# component class that is supposed to wrap around a Lua Table. Inside the Lua Table is supposed to be several methods and variables that can be accessed from both camps. But because Lua(right now) can not inherit methods and variables from C# classes, I can not share methods and variables that were on the C# wrapper.

The "Code" you see on my original post was to bring into existence(on the Lua side) a instance of the table I want to wrap. But just Methods, because I want resources to be made and acquired via a "OnStart" method. That is why you will see "ScriptFile.String()", I wrote a class that will read and potentially write to a file. In short, I only want the user to declare methods to a table they never get to initialize or define, and this will make up a component in a game.

I also made the code create an instance that would not sit on the Global table, it would be returned to C#, and it worked. The bit that still does not work is the meta table that has an Index to a "ScriptingComponent" C# instance. Why do I need this? So I can pass the table reference to various methods as the first argument, I can only best describe it as calling a table method. But none of the C# methods and variables can be access from the Lua side. I'm actually very irritated about this after a week of trying to find something to fix this and still have not found a feasible solution.

I think I might have to force the user to write in only in C# or some other CLI language if they want to define a new game component. I had tons of great ideas on how I was going to use Lua in my game engine but I guess it will never happen unless I fix this.

So I hope this extra (perhaps repeated) information can be of any use at diagnosing my issue.
Thanks again.
